lib/api/objectPut.js

```diff
@@ -45,10 +44,19 @@ function _storeIt(bucketName, objectKey, objMD, authInfo, canonicalID,
         authInfo,
         metaHeaders,
         size,
-        contentType,
+        contentType: request.headers['content-type'],
+        cacheControl: request.headers['cache-control'],
+        contentDisposition: request.headers['content-disposition'],
+        expires: request.headers.expires,
         headers,
         log,
     };
+    // Remove V4 streaming value 'aws-chunked' if present in content-encoding to
```

Comment:
I would extract this into a utility function since it is used 3 times.  Also, it would probably be helpful to put this link in the comments so we have the reference: http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-streaming.html